packages/clerk-js/src/core/clerk.ts

```diff
@@ -1859,24 +1852,44 @@ export class Clerk implements ClerkInterface {
       retries++;
 
       try {
-        const [environment, client] = await Promise.all([
-          Environment.getInstance().fetch({ touch: shouldTouchEnv }),
-          Client.getInstance().fetch(),
-        ]);
+        const initEnvironmentPromise = Environment.getInstance()
+          .fetch({ touch: shouldTouchEnv })
+          .then(res => {
+            this.updateEnvironment(res);
+          });
+
+        const initClient = () => {
+          return Client.getInstance()
+            .fetch()
+            .then(res => this.updateClient(res));
+        };
+
+        const initComponents = () => {
+          if (Clerk.mountComponentRenderer && !this.#componentControls) {
+            this.#componentControls = Clerk.mountComponentRenderer(
+              this,
+              this.environment as Environment,
+              this.#options,
+            );
+          }
+        };
+
+        await Promise.all([initEnvironmentPromise, initClient()]).catch(async e => {
+          // limit the changes for this specific error for now
+          if (isClerkAPIResponseError(e) && e.errors[0].code === 'requires_captcha') {
```

Comment:
This could be a util under `@clerk/shared/error`, as it is used in numerous places. We already have similar utils, like `is4xxError` etc.